### PR TITLE
stats: correct fast analyze stats caclulation

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

The stats built from fast analyze were wrong:

- The `TotColSize` was the column size in the sample, not the `total` column size.
- When built CM Sketch with top N, we scale each value by `rowCount/sampleSize`, even if the value just occurred once, which will overestimate the row count, and it also conflicts with the calculation of `defaultValue`, which treat the only once items the same as the values that do not occur.

### What is changed and how it works?

- Scale the total column size by `rowCount/sampleSize`.
- For values that only occurred once in the sample, it uses `defaultValue` as their row count.
### Check List 

Tests 

 - Unit test
Code changes

 - Has exported function/method change

Side effects

 - None

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: statistics/cmsketch.go [hunk 1]
```diff
@@ -116,37 +116,43 @@ func NewCMSketchWithTopN(d, w int32, sample [][]byte, numTop uint32, rowCount ui
...
+			c.topN[h1] = append(c.topN[h1], &TopNMeta{h2, data, rowCount})
 		} else {
-			c.insertBytesByCount(data, scaledCount)
+			c.insertBytesByCount(data, rowCount)
 		}
 	}
 	return
 }
 
-func (c *CMSketch) calculateDefaultVal(helper *topNHelper, estimateNDV, scaleRatio, rowCount uint64) {
```
Comment (winoros) [inline statistics/cmsketch.go]: Any reason why change this method?

Comment (lamxTyler) [reply statistics/cmsketch.go]: We need the default value before constructing cm sketch, so change it to return values.

File: executor/analyze.go [hunk 2]
```diff
@@ -1001,6 +1001,8 @@ func (e *AnalyzeFastExec) buildHist(ID int64, collector *statistics.SampleCollec
 	// build CMSketch
 	var ndv, scaleRatio uint64
 	collector.CMSketch, ndv, scaleRatio = statistics.NewCMSketchWithTopN(defaultCMSketchDepth, defaultCMSketchWidth, data, 20, uint64(rowCount))
+	// Scale the total column size.
+	collector.TotalSize *= rowCount / int64(len(collector.Samples))
```
Comment (eurekaka) [inline executor/analyze.go]: This `TotalSize` computed in `CalcTotalSize` contains the flag byte, which is inconsistent with normal analyze as:
```golang
func (c *SampleCollector) collect(sc *stmtctx.StatementContext, d types.Datum) error {
    if !c.IsMerger {
        if d.IsNull() {
            c.NullCount++
            return nil
        }
        // Minus one is to remove the flag byte.
        c.TotalSize += int64(len(d.GetBytes()) - 1)
    }
```

Comment (lamxTyler) [reply executor/analyze.go]: Values here are already decoded, there are no flags in it.

**Pull request merged**